modules/docs/arrow-docs/docs/docs/typeclasses/traverse/README.md

```diff
@@ -9,8 +9,259 @@ permalink: /docs/typeclasses/traverse/
+}
+
+For our example, F would be `List` (the initial container), G would be the data type representing the side effect: `Option`, `Either` or `DeferredK`.
+
+So, if we have a `List<User>` (the ones we want to obtain their profiles) and a function `User -> DeferredK<Profile>`, with traverse we can transform and instead of obtaining a `List<DeferredK, Profile>` it can aggregate all the results to obtain a single `DeferredK<List,Profile>`.
+
+In this case, `traverse` can go over the collection, apply the function and aggregate the resulting values (with side effects) in a `List`.
+
+Basically, `F` is some sort of context which may contain a value. We are using List in the example, but there are `Traverse` implementations for `Option`, `Either` or `Validated`.
+
+Let's see another example further clarifying this.
+
+```kotlin:ank
+fun parseIntEither(s: String): Either<NumberFormatException, Int> =
+  Either.applicativeError<NumberFormatException>().catch(
+    { NumberFormatException("Error converting $s to Int") },
+    { s.toInt() }
+  ).fix()
+
+fun parseIntValidated(s: String): ValidatedNel<NumberFormatException, Int> =
+  Validated.fromEither(parseIntEither(s)).toValidatedNel()
+```
+
+We can use these two functions to traverse a collection containing strings, converting them to integers and accumulating the errors with Either or Validated.
+
+Example of what these functions do:
+
+```kotlin:ank
+parseIntEither("1")
+parseIntEither("jimmy")
+```
+
+Examples of going through a list with `map` and with `traverse` for `Either`:
+
+```kotlin:ank
+val listOfValidNumbers = listOf("1", "2", "3")
+val listOfInvalidNumbers = listOf("1", "jimmy", "peter")
+
+listOfValidNumbers.map { s -> parseIntEither(s) }
+
+listOfValidNumbers.traverse(
+  Either.applicative()
+) { element -> parseIntEither(element) }
+
+listOfInvalidNumbers.traverse(
+  Either.applicative()
+) { element -> parseIntEither(element) }
+```
+
+Examples of going through the list with `traverse` for `Validated`:
+
+```kotlin:ank
+listOfValidNumbers.traverse(
+  ValidatedNel.applicative(Nel.semigroup<NumberFormatException>())) { element -> parseIntValidated(element)}
+
+listOfInvalidNumbers.traverse(
+  ValidatedNel.applicative(Nel.semigroup<NumberFormatException>())) { element -> parseIntValidated(element)}
+```
+
+Let's explain what's going on here when traversing a `list` with `Validated`.
+We are using an `Applicative` instance of `ValidatedNel`, and for that we need to provide a "proof" that the non-empty-list (Nel) is a `SemiGroup`.
+The `Applicative` typeclass instance for `ValidatedNel` allows to run independent computations. And the `SemiGroup` typeclass instance allows us to combine elements of the same type, in this case, it helps `ValidatedNel` with the task of accumulating the errors.
+
+If you want to see other example, you could visit: https://www.enhan.eu/how-to-in-fp/
+
+
+## Sequence
```

Comment:
`Sequence` is something else. When referring to combinators always use their actual name. In this case `sequence`

Reply:
changed